docs: create activities and remote script execution explanation doc (#77)
canonical/landscape-documentation: docs/explanation/features/remote-script-execution.md

(explanation-remote-script-execution)=

> # Landscape remote script execution

Reviewer: ```suggestion
# Remote script execution
```
I try to keep "Landscape" out of the title when possible (although sometimes it makes sense to include it). I think with the suggested context I added to the intro, it isn't necessary to specify here - it's clear enough that the doc refers to the specific feature in Landscape, and not just the general topic of "script execution"

## Attachments

Scripts may include attachments. Attachments are stored on Landscape Server, and clients can fetch them before execution.

## `execute-script` message

Landscape Server sends a message to Landscape Client in the following form:

```json
{
  "type": "execute-script",
  "interpreter": "INTERPRETER",
  "code": "SCRIPT_CODE",
  "username": "USER",
  "time-limit": "TIME_LIMIT",
  "operation-id": "ACTIVITY_ID",
  "attachments": ["ATTACHMENT_IDS"],
  "env": {
    "LANDSCAPE_ACCOUNT": "ACCOUNT_NAME",
    "LANDSCAPE_COMPUTER_ID": "COMPUTER_ID",
    "LANDSCAPE_COMPUTER_TAGS": ["COMPUTER_TAGS"],
    "LANDSCAPE_URL": "LANDSCAPE_URL",
    "LANDSCAPE_ACTIVITY_ID": "ACTIVITY_ID",
    "LANDSCAPE_ACTIVITY_CREATOR_ID": "CREATOR_ID",
    "LANDSCAPE_ACTIVITY_CREATION_TIME": "CREATION_TIME"
  }
}
```

Field descriptions:

`interpreter`: The interpreter to run the script. This is parsed from the script code.
`code`: The script body, excluding the interpreter line.

> `username`: The user under which the script will run. In the Landscape Client snap, this is always root.

Reviewer: ```suggestion
`username`: The user under which the script will run. If you're using the Landscape Client snap, this is always root.
```
OR, you could specify what the user is for the deb too (e.g., "For the Landscape Client deb, the administrator sets the user. For the snap, the user is always root.")

Just suggesting this wording change because wording it as "In the Landscape client snap" makes it feel more like that's the main version of Client (IMO at least)

> `interpreter`: The interpreter to run the script. This is parsed from the script code.
> `code`: The script body, excluding the interpreter line.
> `username`: The user under which the script will run. In the Landscape Client snap, this is always root.
> `time_limit`: Maximum execution time before the process is forcibly terminated.
> `activity_id`: Unique identifier of the activity.
> `attachments`: List of attachment IDs stored on Landscape Server.
> `env`: Environment variables provided by Landscape Server, including account name, computer metadata, and activity metadata.

Reviewer: ```suggestion
- `interpreter`: The interpreter to run the script. This is parsed from the script code.
- `code`: The script body, excluding the interpreter line.
- `username`: The user under which the script will run. In the Landscape Client snap, this is always root.
- `time_limit`: Maximum execution time before the process is forcibly terminated.
- `activity_id`: Unique identifier of the activity.
- `attachments`: List of attachment IDs stored on Landscape Server.
- `env`: Environment variables provided by Landscape Server, including account name, computer metadata, and activity metadata.
```
Otherwise it gets formatted like a paragraph
